addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -191,10 +191,21 @@ def _prepare_iap_payload(self):
         """
         self.ensure_one()
         payload = {'lead_number': self.lead_number,
-                   'search_type': self.search_type,
-                   'countries': self.country_ids.mapped('code')}
-        if self.state_ids:
-            payload['states'] = self.state_ids.mapped('code')
+                   'search_type': self.search_type}
+        countries = []
+        
+        for country in self.country_ids:
+            j_country = {}
+            j_country['code'] = country.code
+            if self.state_ids:
+                country_states = []
+                for state_id in self.state_ids:
+                    if state_id in country.state_ids:
+                        country_states.append(state_id.code)
+                j_country['states'] = country_states
```

Comment:
j_country['state'] = self.state_ids.filtered(lambda state: state in country.state_ids).mapped('code')